core/src/main/java/com/graphhopper/routing/DefaultWeightingFactory.java

```diff
@@ -55,6 +57,15 @@ public Weighting createWeighting(Profile profile, PMap requestHints, boolean dis
         hints.putAll(requestHints);
 
         final String vehicle = profile.getVehicle();
+        // todonow: we used to take this information from the 'encoder', but now it is gone. We probably need to take
+        // it from the profile now somehow.
+        final boolean isMotorVehicle = true;
+        if (isMotorVehicle) {
```

Comment:
Either put it into the weighting string like `weighting: fastest|private_factor=10` or into the profile like `weighting: fastest, private_factor: 10`. Or somehow map the vehicle/profile name to isMotorVehicle...

Reply:
This seems to be the last open question for me here.

Reply:
Maybe the least invasive thing we can do is use a list of vehicle names that we consider `!isMotorVehicle` like `foot,hike,wheelchair,bike,mtb,racingbike,bike2`? This is a bit ugly, because the list needs to be adjusted if someone adds a new 'vehicle', but this requires adjusting the flag encoder factor also, anyway and we can leave a comment there.

Reply:
Would it be possible to add a new vehicle without forking GraphHopper then? With the old FlagEncoder approach this was still doable.

Reply:
Yes, absolutely. This is just about the private/destination penalty in FastestWeighting. And even that can be adjusted by overwriting the DefaultWeightingFactory. Let alone that the private/destination penalty should be a thing of the past as we have much more possibilities using CustomWeighting.

... then again if you have a certain use-case in mind it would also help if you tried this branch to make sure I did not mess this up somehow :)

Reply:
You can even change the private/destination penalty using the hints in the profile definition, i.e. in `config.yml`.